libs/designsystem/tab-navigation/src/tab-navigation/tab-navigation.component.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  QueryList,
+  ViewChild,
+} from '@angular/core';
+import { filter, fromEvent, map, Subject, takeUntil, tap } from 'rxjs';
+import { WindowRef } from '@kirbydesign/designsystem/types';
+import { TabNavigationItemComponent } from '../tab-navigation-item/tab-navigation-item.component';
+
+const ARROW_LEFT = 'ArrowLeft';
+const ARROW_RIGHT = 'ArrowRight';
+
+@Component({
+  selector: 'kirby-tab-navigation',
+  templateUrl: './tab-navigation.component.html',
+  styleUrls: ['./tab-navigation.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TabNavigationComponent implements AfterViewInit, OnDestroy {
+  public readonly DEBOUNCE_TIME_MS = 250;
+
+  @ViewChild('tabBar')
+  private tabBar: ElementRef<HTMLElement>;
+
+  @ContentChildren(TabNavigationItemComponent, { read: ElementRef })
+  private tabs: QueryList<ElementRef<HTMLElement>>;
+
+  private tabBarElement: HTMLElement;
+  private tabElements = new Array<HTMLElement>();
+  private tabButtonElements = new Array<HTMLElement>();
+  private destroyed$ = new Subject<void>();
+
+  @Input()
+  get selectedIndex(): number {
+    return this._selectedIndex;
+  }
+
+  set selectedIndex(index: number) {
+    if (index !== this._selectedIndex) {
+      this._selectedIndex = index;
+
+      setTimeout(() => {
+        this.scrollToTab(this.selectedIndex);
+        this.focusTab(this.selectedIndex);
+        this.selectTab(this.selectedIndex);
+        this.selectedIndexChange.emit(this.selectedIndex);
+      });
+    }
+  }
+  private _selectedIndex = -1;
+
+  @Output()
+  selectedIndexChange = new EventEmitter<number>();
+
+  constructor(private window: WindowRef) {
+    /**/
+  }
+
+  ngAfterViewInit(): void {
+    this.tabBarElement = this.tabBar.nativeElement;
+    this.tabs.forEach((tab) => this.tabElements.push(tab.nativeElement));
+    this.tabElements.forEach((tabElement) =>
+      this.tabButtonElements.push(tabElement.querySelector('[role="tab"]'))
+    );
+    this.initTabListeners();
+    setTimeout(() => {
+      this.selectTab(this.selectedIndex);
+      this.scrollToTab(this.selectedIndex);
+      this.focusTab(this.selectedIndex);
+    }, this.DEBOUNCE_TIME_MS);
+  }
+
+  initTabListeners(): void {
```

Comment:
I think this code seems good and sound, but this approach to listening for events is a bit difficult to understand, and I would suggest to simplify it by using the HostListener decorator from angular that also takes care of removing event listeners etc. so we can avoid cleanup. 

To make what I mean easier to comprehend I have made a small POC - still some loose ends but I hope you can see what I mean by simplifying it a bit: https://github.com/kirbydesign/designsystem/compare/enhancement/2705-kirby-component-from-experimental-local-navigation...tab-nav-listener-refactorhttps://github.com/kirbydesign/designsystem/compare/enhancement/2705-kirby-component-from-experimental-local-navigation...tab-nav-listener-refactor

Reply:
Done